frontend/js/pages/BattleList.js

```diff
@@ -38,25 +42,37 @@ function BattleList(props) {
       <ul className="list_battle">
         <div className="settled">
           <h3>Settled battles</h3>
-          {battles.map((battle) =>
+          {battleArray.map((battle) =>
             battle.winner ? (
               <li key={battle.id} className="item">
-                <a className="battle_settled" href={Urls.battle_detail_v2(battle.id)}>
+                <Link
+                  className="battle_settled"
+                  to={{
+                    pathname: Urls.battle_detail_v2(battle.id),
+                    query: { battle, pokemons, battleUsers },
```

Comment:
I see here what you were doing before, an ingenious solution, however, we make it better and simple! Notice, you are persisting to the location information you already have available on the store, you should in your battle detail get this information from the store using `mapStateToProps` and `denormalize` as I mentioned before

Reply:
Thank you!